src/Widgets/TaskRow.vala

```diff
@@ -130,8 +118,49 @@ public class Tasks.TaskRow : Gtk.ListBoxRow {
         });
 
         key_release_event.connect ((event) => {
-            if (event.keyval == Gdk.Key.space) {
-                check.active = !check.active;
+            switch (event.keyval) {
+                case Gdk.Key.space:
+                    if (has_focus) {
+                        check.active = !check.active;
+                    }
+                    break;
+
+                case Gdk.Key.Return:
+                case Gdk.Key.KP_Enter:
+                    if (has_focus) {
+                        show_popover ();
+                    } else {
+                        grab_focus ();
+                    }
+                    break;
+
+                case Gdk.Key.BackSpace:
+                    if (has_focus) {
+                        // TODO: Delete Task
```

Comment:
In general, I would prefer if we didn't merge any dead code. We can use GitHub issues to describe ToDo's :)

Reply:
If I got this straight you mean I should remove the `case Gdk.Key.BackSpace: ...` from the PR, right?

Reply:
Dropped the case for BackSpace.